packages/trader/src/AppV2/Components/TradeParameters/Stake/stake-input.tsx

```diff
@@ -0,0 +1,330 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+
+import { formatMoney, getCurrencyDisplayCode, getDecimalPlaces } from '@deriv/shared';
+import { Localize, localize } from '@deriv/translations';
+import { ActionSheet, TextFieldWithSteppers } from '@deriv-com/quill-ui';
+
+import { useDtraderQuery } from 'AppV2/Hooks/useDtraderQuery';
+import { getPayoutInfo, getProposalRequestObject } from 'AppV2/Utils/trade-params-utils';
+import { getDisplayedContractTypes } from 'AppV2/Utils/trade-types-utils';
+import { ExpandedProposal, getProposalInfo } from 'Stores/Modules/Trading/Helpers/proposal';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { TTradeStore } from 'Types';
+
+import StakeDetails from './stake-details';
+
+type TOnProposalResponse = TTradeStore['onProposalResponse'];
+type TStakeInput = {
+    onClose: () => void;
+    is_open?: boolean;
+};
+type TNewValues = {
+    amount?: string | number;
+    payout_per_point?: string | number;
+    barrier_1?: string | number;
+};
+
+const StakeInput = observer(({ onClose, is_open }: TStakeInput) => {
+    const trade_store = useTraderStore();
+    const {
+        amount,
+        commission,
+        contract_type,
+        currency,
+        barrier_1,
+        has_stop_loss,
+        is_accumulator,
+        is_multiplier,
+        is_turbos,
+        is_vanilla,
+        onChange,
+        proposal_info,
+        stop_out,
+        trade_type_tab,
+        trade_types,
+        validation_params,
+    } = trade_store;
+
+    const [proposal_request_values, setNewValues] = React.useState<TNewValues>({ amount }); // contains information for creating proposal request: stake (amount), payout_per_point for Turbos and barrier_1 for Vanillas
+    const [stake_error, setStakeError] = React.useState('');
+    const [fe_stake_error, setFEStakeError] = React.useState('');
+
+    const contract_types = getDisplayedContractTypes(trade_types, contract_type, trade_type_tab);
+    const should_show_payout_details = !is_accumulator && !is_multiplier && !is_turbos && !is_vanilla;
+
+    const {
+        contract_payout: first_contract_payout,
+        max_payout,
+        error: first_payout_error,
+    } = getPayoutInfo(proposal_info[contract_types[0]]);
```

Comment:
Created `getPayoutInfo` function for information extraction, you can check it in trade-params-utils.tsx file below. It contains some scary hardcoded stuff, that Maryia did and we will need to get rid of, when API will be updated